### ISSUE TITLE
Missing swiftlint arguments when passing in a custom executable path

### ISSUE THREAD

**Issue by nap-sam-dean** [fastlane/fastlane#9533]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description

Running swiftlint with a custom executable will fail to version check some optional parameters. These include `--strict` and `--quiet`.

I have identified the issue, and PR is incoming - it's a simple reference / value passing issue when constructing the swiftlint command with a custom executable parameter.

##### Complete output when running fastlane, including the stack trace and command used
> You can use: `--capture_output` as the last commandline argument to get that collected for you

Command:

FastFile:
```
  # LINT
  lane :lint do |args|
    mode = args[:mode] || "lint"
    strict = mode == "lint"

    swiftlint(
      executable: File.expand_path('../Client/Pods/SwiftLint/swiftlint'),
      mode: mode,
      output_file: 'fastlane/swiftlint.txt',
      config_file: 'fastlane/swiftlint.yml',
      strict: strict
    )
  end
```

Command ran: 

`> bundle exec fast lane ios lint`

Output:

```
...
Unrecognized arguments: "version"
[18:28:22]: Your version of swiftlint (0) does not support '--strict' option.
Update swiftlint to 0.9.2 or above using `brew update && brew upgrade swiftlint`
The option will be ignored.
[18:28:22]: {correct path}/swiftlint lint --config fastlane/swiftlint.yml
...
```

Expected output:

```
...
[18:28:22]: {correct path}/swiftlint lint --strict --config fastlane/swiftlint.yml
...
```

Note - no warning, and `--strict` is passed as an argument. This is also the output when ran using the brew install of swiftlint and no `:executable` parameter in the FastFile.

### Environment

Please run `fastlane env` and copy the output below. This will help us help you :+1:
If you used `--capture_output` option please remove this block - as it is already included there.

```
<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.12.4                                     |
| Ruby                        | 2.0.0                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.11.0 (Apple Git-81)           |
| Installation Source         | ~/.rbenv/versions/2.0.0-p648/bin/fastlane   |
| Host                        | Mac OS X 10.12.4 (16E195)                   |
| Ruby Lib Dir                | ~/.rbenv/versions/2.0.0-p648/lib            |
| OpenSSL Version             | OpenSSL 1.0.2k  26 Jan 2017                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 8.3.3                                       |


### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |


### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
skip_docs

platform :ios do

  travisTestiOS = ENV["CI_IOS"]
  testDevice = ENV["CI_DEVICE"] || "iPhone 7 (10.3)"

  before_all do
    ENV["FASTLANE_EXPLICIT_OPEN_SIMULATOR"] = "1"
  end

  ###

  # BUILD CLIENT DEV
  lane :client_build_dev do
    xcodebuild(
      workspace: "Client/Client.xcworkspace",
      scheme: "Client",
      configuration: "Debug",
      # clean: options[:clean],
      destination: "generic/platform=iOS",
      build_settings: {
        "CODE_SIGNING_REQUIRED" => "NO",
        "CODE_SIGN_IDENTITY" => ""
      }
    )
  end

  # BUILD CLIENT TEST FLIGHT
  lane :client_testflight_archive do
	  gym(
      clean: true,
      workspace: 'Client/Client.xcworkspace',
      scheme: 'ClientTestflight',
      codesigning_identity: 'iPhone Distribution: Net-A-Porter International Limited',
      output_directory: './build',
      output_name: 'YNSDK-iOS-Client'    
	  )
  end

  # LOCAL SCAN CLIENT
  lane :client_local_test do
    scan(
      skip_build: true,
      workspace: "Client/Client.xcworkspace",
      scheme: "Client",
      device: "#{testDevice}",
      clean: false
    )
  end

  # TRAVIS SCAN CLIENT
  lane :client_travis_test do
    scan(
      skip_build: true,
      workspace: "Client/Client.xcworkspace",
      scheme: "Client",
      device: "#{testDevice}",
      clean: false
    )
  end

  # LINT
  lane :lint do |args|
    mode = args[:mode] || "lint"
    strict = mode == "lint"

    swiftlint(
      executable: File.expand_path('../Client/Pods/SwiftLint/swiftlint'),
      mode: mode,
      output_file: 'fastlane/swiftlint.txt',
      config_file: 'fastlane/swiftlint.yml',
      strict: strict
    )
  end

  # Distribution
  lane :client_distribution do
  	pilot(
	    username: "sam.dean@net-a-porter.com",
	    team_name: "Net-A-Porter International Limited",
	    ipa: "build/YNSDK-iOS-Client.ipa",
	    skip_submission: true,
	    skip_waiting_for_build_processing: true
	  )
  end

  # PRIVATE INIT CERTIFICATE
  desc "setup the signing certificates and provisioning profiles"
  private_lane :initCertificates do
	  # add the certificates and keys to the keychain and use fastlane-credentials for the pilot step
	  sh("./certificates/add-key.sh")
	  sh("fastlane fastlane-credentials add --username $ITUNESCONNECT_USERNAME --password $ITUNESCONNECT_PASSWORD")
	  sh("security list-keychains")
	  # this step is done to allow the keychain to be used when signing
	  sh("security set-key-partition-list -S apple-tool:,apple: -s -k travis ios-build.keychain-db")
	  # update the CFBundleVersion to the number of commits
	  sh("./setBuildVersion.sh")
  end

  ### ### ###

  # LOCAL DEFAULT
  desc "build-test-lint"
  lane :default do
    client_local_test
    lint
  end
    
  # TRAVIS DEFAULT
  desc "build-test-lint"
  lane :travisDefault do
    reset_simulators( ios: ["{#travisTestiOS}"] )
    client_travis_test
    lint
  end
  
  # TESTFLIGHT BUILD
  desc "build-archive-test-distribution"
  lane :testflightBuild do
    initCertificates
	  client_testflight_archive
    client_travis_test
	  client_distribution
    # clean up the keychain
    sh("./certificates/remove-key.sh")
  end

end
```
</details>

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.39.2  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| slack-notifier            | 1.5.1        |
| CFPropertyList            | 2.3.5        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.5.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.8        |
| commander-fastlane        | 4.4.4        |
| faraday-cookie_jar        | 0.0.6        |
| gh_inspector              | 1.0.3        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.1        |
| os                        | 0.9.6        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.12.0       |
| mini_magick               | 4.5.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| json                      | 2.1.0        |
| excon                     | 0.57.0       |
| rubyzip                   | 1.2.1        |
| plist                     | 3.3.0        |
| faraday                   | 0.12.1       |
| signet                    | 0.7.3        |
| faraday_middleware        | 0.11.0.1     |
| http-cookie               | 1.0.3        |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| fastimage                 | 2.1.0        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.0        |
| bundler                   | 1.15.1       |
| terminal-table            | 1.8.0        |
| unicode-display_width     | 1.3.0        |
</details>

*generated on:* **2017-06-20**
</details>
```

**Comment by nap-sam-dean** [comment]
Fixed in 2.42.0 - https://github.com/fastlane/fastlane/releases/tag/2.42.0

**Comment by mpirri** [comment]
@nap-sam-dean You okay with closing this PR if this is fixed?  Thanks much!  🚀
